rest_sensors.cpp

```diff
@@ -803,14 +803,7 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
 
                 if (rid.suffix == RConfigDeviceMode) // String
                 {
-                    if (RConfigDeviceModeValues.indexOf(data.string) >= 0)
-                    {
-                        pendingMask |= R_PENDING_DEVICEMODE;
-                        sensor->enableRead(WRITE_DEVICEMODE);
-                        sensor->setNextReadTime(WRITE_DEVICEMODE, QTime::currentTime());
-                        updated = true;
-                    }
-                    else if (!data.string.isEmpty())
+                    if (!data.string.isEmpty() && RConfigDeviceModeValues.indexOf(data.string) >= 0)
```

Comment:
The if statement could be deleted if the `isEmpty()` check is moved one level up.